File: backend/dto/src/main/java/pl/dawid0604/pcForum/dto/post/ExtendedPostDTO.java
package pl.dawid0604.pcForum.dto.post;

import pl.dawid0604.pcForum.dto.user.UserProfileDTO;

import java.util.List;

public record ExtendedPostDTO(String encryptedId, UserProfileDTO user, String createdAt,
                              Thread thread, List<PostContentDTO> content) {

    public record Thread(String encryptedId, String title) { }
}

File: frontend/src/app/model/ExtendedPostDTO.ts
import { PostContentDTO } from "./PostContentDTO"
import { UserProfileDTO } from "./UserProfileDTO"

export interface ExtendedPostDTO {
    encryptedId: string,
    user: UserProfileDTO,
    createdAt: string,
    thread: Thread,
    content: PostContentDTO[]
}

export interface Thread {
    encryptedId: string,
    title: string
}